api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -662,6 +662,9 @@ public ResponseEntity<WorkspaceUserRolesResponse> shareWorkspace(
       throw new BadRequestException("Missing required update field 'workspaceEtag'");
     }
 
+    workspaceAuthService.enforceWorkspaceAccessLevel(
+        workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER);
```

Comment:
I misspoke earlier. Share is owner only: https://cs.github.com/all-of-us/workbench/blob/50d09d4e4fac8a1b8816368235bcb9d997a726c3/api/src/main/java/org/pmiops/workbench/firecloud/FirecloudTransforms.java?q=canshare#L37

Also please remove the comment around L725